src/order/sup_indep.lean

```diff
@@ -0,0 +1,127 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.lattice
+import data.set.pairwise
+
+/-!
+# Finite supremum independence
+
+In this file, we define supremum independence of indexed sets. An indexed family `f : ι → α` is
+sup-independent if, for all `a`, `f a` and the supremum of the rest are disjoint.
+
+In distributive lattices, this is equivalent to being pairwise disjoint. Note however that
+`set.pairwise_disjoint` does not currently handle indexed sets.
```

Comment:
This is obsolete since #9898, right?

Reply:
Yup, I was bumping master literally just now!